File: src/test/java/ransomnote/SolutionTest.java
package ransomnote;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {

    private final Solution solution = new Solution();

    @Test
    public void testOne() {
        String ransomNote = "aa";
        String magazine = "aab";

        boolean canConstruct = solution.canConstruct(ransomNote, magazine);

        assertThat(canConstruct).isTrue();
    }

    @Test
    public void testTwo() {
        String ransomNote = "a";
        String magazine = "b";

        boolean canConstruct = solution.canConstruct(ransomNote, magazine);

        assertThat(canConstruct).isFalse();
    }

    @Test
    public void testThree() {
        String ransomNote = "aa";
        String magazine = "ab";

        boolean canConstruct = solution.canConstruct(ransomNote, magazine);

        assertThat(canConstruct).isFalse();
    }
}

File: src/test/kotlin/dev/sanket/maxdepthofbtree/SolutionTest.kt
package dev.sanket.maxdepthofbtree

import assertk.assertThat
import assertk.assertions.isEqualTo
import assertk.assertions.isZero
import dev.sanket.dev.sanket.common.buildTree
import dev.sanket.dev.sanket.maxdepthofbtree.Solution
import org.junit.jupiter.api.Test

class SolutionTest {

    private val solution = Solution()

    @Test
    fun shouldReturnZeroWhenRootIsNull() {
        assertThat(solution.maxDepth(null)).isZero()
    }

    @Test
    fun testOne() {
        val root = buildTree(intArrayOf(3, 9, 20, 0, 0, 15, 7), 0)

        assertThat(solution.maxDepth(root)).isEqualTo(3)
    }
}